pkg/provider/docker/docker.go

```diff
@@ -176,7 +180,7 @@ func (p *Provider) getClientOpts() ([]client.Opt, error) {
 			return nil, err
 		}
 
-		opts = append(opts, client.WithHTTPClient(&http.Client{Transport: tr}))
+		opts = append(opts, client.WithHTTPClient(&http.Client{Transport: tr, Timeout: time.Duration(p.HTTPClientTimeout)}))
```

Comment:
It seems that the timeout is not needed here, since it will be set to the client anyway because of L159 ?

```suggestion
		opts = append(opts, client.WithHTTPClient(&http.Client{Transport: tr}))
```

Reply:
the timeout must be defined here because of the order in which the options are applied.

Reply:
Ok i agree, i made a wrong assumption